opm/simulators/wells/BlackoilWellModelConstraints.cpp

```diff
@@ -561,8 +561,9 @@ template<class Scalar>
 bool BlackoilWellModelConstraints<Scalar>::
 updateGroupIndividualControl(const Group& group,
                              const int reportStepIdx,
-                             std::map<std::pair<std::string,Phase>,std::string>& switched_inj,
-                             std::map<std::string, std::string>& switched_prod,
+                             const int max_number_of_group_switch,
+                             std::map<std::pair<std::string,Phase>,std::vector<std::string>>& switched_inj,
```

Comment:
`Map<pair, vector>` really isn't a good data structure for anything.  I suppose we'd create a separate type to handle some of the logic here, but barring that I'd probably either pull the `Phase` out of the pair and make it a `vector<map<string, vector>>`, or pull the group name out and make it a `map<string, vector<vector>>`.  The former would  be indexed, at least conceptually, as `switched_inj[phase][group]` while the latter would be indexed as `switched_inj[group][phase]`.  Given the current loop structure I think the latter may be preferable.

On a second note, since we now **know** that we're dealing with injection controls we should store the integer [`Group::InjectionCMode`](https://github.com/OPM/opm-common/blob/268f5b8b3e5cb89a7d304f4d31bc6fdf4dd96dc9/opm/input/eclipse/Schedule/Group/Group.hpp#L70) values instead of their string representation.  That way we use less memory and the comparisons are more efficient as well.

If we use the group name as the primary key and the `Phase` as the secondary key I'd probably write this block along the lines of
```C++
bool group_is_oscillating = false;
if (auto groupPos = switched_inj.find(group.name()); groupPos != switched_inj.end()) {
    auto& ctrls = groupPos->second[static_cast<std::underlying_type_t<Phase>>(phase)];
    for (const auto& ctrl : ctrls) {
        if (std::count(ctrls.begin(), ctrls.end(), ctrl) < max_number_of_group_switches) {
            continue;
        }

        if (ctrls.back() != *(ctrls.end() - 2)) {
            if (rank == 0) {
                // report, use InjectionCMode2String() for string value of 'ctrl'
            }
            ctrls.push_back(ctrl);
        }
        group_is_oscillating = true;
        break;
    }
}
```